scripts/translate_doc_pr/section_matcher.py

```
@@ -0,0 +1,973 @@
+    
+    thread_safe_print(f"   🔍 Searching for direct matches among {len(target_headers)} target headers")
+    
+    for source_line_num, source_hierarchy in source_sections.items():
+        # Extract the leaf title from hierarchy
+        source_leaf_title = source_hierarchy.split(' > ')[-1] if ' > ' in source_hierarchy else source_hierarchy
+        source_clean_title = clean_title_for_matching(source_leaf_title)
+        
+        thread_safe_print(f"   📝 Looking for match: {source_clean_title}")
+        
+        if not is_system_variable_or_config(source_leaf_title):
+            thread_safe_print(f"      ⚠️  Not a system variable/config, skipping direct match")
```

Ruff (0.13.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)